clustered/client/src/main/java/org/ehcache/clustered/client/internal/config/xml/ClusteringCacheManagerServiceConfigurationParser.java

```diff
@@ -0,0 +1,466 @@
+  }
+
+  @Override
+  public Class<ClusteringService> getServiceCreationType() {
+    return ClusteringService.class;
+  }
+  /**
+   * Translates a {@link ServiceCreationConfiguration} to an xml element
+   *
+   * @param doc
+   *
+   * @param serviceCreationConfiguration
+   */
+  @Override
+  public Node marshalServiceCreationConfiguration(final Document doc, final ServiceCreationConfiguration<ClusteringService> serviceCreationConfiguration) {
+    validateParametersForTranslationToServiceConfig(doc, serviceCreationConfiguration);
+    Element rootElement;
+    ClusteringServiceConfiguration clusteringServiceConfiguration = (ClusteringServiceConfiguration)serviceCreationConfiguration;
+    rootElement = createConnectionElement(doc, clusteringServiceConfiguration);
+    processTimeUnits(doc, rootElement, clusteringServiceConfiguration);
+    Element serverSideConfigurationElem = processServerSideElements(doc, clusteringServiceConfiguration);
+    rootElement.appendChild(serverSideConfigurationElem);
+    return rootElement;
+  }
+
+  private void validateParametersForTranslationToServiceConfig(Document doc
+    , ServiceCreationConfiguration<ClusteringService> serviceCreationConfiguration) {
+    Objects.requireNonNull(doc, "Document must not be NULL");
+    Objects.requireNonNull(serviceCreationConfiguration, "ServiceCreationConfiguration must not be NULL");
+    if (!(serviceCreationConfiguration instanceof ClusteringServiceConfiguration)) {
+      throw new IllegalArgumentException("Parameter serviceCreationConfiguration must be of type ClusteringServiceConfiguration."
+                                         + "Provided type of parameter is : " + serviceCreationConfiguration.getClass());
+    }
+  }
+
+  private Element createRootUrlElement(Document doc, ClusteringServiceConfiguration clusteringServiceConfiguration) {
+    Element rootElement = doc.createElement(TC_CLUSTERED_NAMESPACE_PREFIX + COLON + CLUSTER_ELEMENT_NAME);
+    rootElement.setAttributeNS(XMLConstants.XMLNS_ATTRIBUTE_NS_URI, "xmlns:" + TC_CLUSTERED_NAMESPACE_PREFIX, NAMESPACE.toString());
+    Element urlElement = doc.createElement(TC_CLUSTERED_NAMESPACE_PREFIX + COLON + CONNECTION_ELEMENT_NAME);
+    urlElement.setAttribute(URL_ATTRIBUTE_NAME, clusteringServiceConfiguration.getClusterUri().toString());
+    rootElement.appendChild(urlElement);
+    return rootElement;
+  }
+
+  private Element createServerElement(Document doc, ClusteringServiceConfiguration clusteringServiceConfiguration) {
+    if (!(clusteringServiceConfiguration.getConnectionSource() instanceof ConnectionSource.ServerList)) {
+      throw new IllegalArgumentException("When connection URL is null, source of connection MUST be of type ConnectionSource.ServerList.class");
+    }
+    ConnectionSource.ServerList servers = (ConnectionSource.ServerList)clusteringServiceConfiguration.getConnectionSource();
+    if (!servers.getServers().iterator().hasNext()) {
```

Comment:
Can't we let the xml schema validate this for us?

Reply:
Same as above. 
I wanted to make explicit check so that I can throw an exception with custom message, instead a generic parser validation error. Let me know if I should depend only on schema validator.

Reply:
For now I would let the existing codification of these rules enforce them rather than duplicating that information.